jax_galsim/interpolatedimage.py

```diff
@@ -0,0 +1,1115 @@
+            major, minor = compute_major_minor_from_jacobian(
+                self._jac_arr.reshape((2, 2))
+            )
+            return self._jax_aux_data["_force_maxk"] * minor
+        else:
+            return self._getMaxK(self._jax_aux_data["calculate_maxk"])
+
+    @lazy_property
+    def _stepk(self):
+        if self._jax_aux_data["_force_stepk"]:
+            major, minor = compute_major_minor_from_jacobian(
+                self._jac_arr.reshape((2, 2))
+            )
+            return self._jax_aux_data["_force_stepk"] * minor
+        else:
+            return self._getStepK(self._jax_aux_data["calculate_stepk"])
+
+    def _getStepK(self, calculate_stepk):
+        # GalSim cannot automatically know what stepK and maxK are appropriate for the
+        # input image.  So it is usually worth it to do a manual calculation (below).
+        #
+        # However, there is also a hidden option to force it to use specific values of stepK and
+        # maxK (caveat user!).  The values of _force_stepk and _force_maxk should be provided in
+        # terms of physical scale, e.g., for images that have a scale length of 0.1 arcsec, the
+        # stepK and maxK should be provided in units of 1/arcsec.  Then we convert to the 1/pixel
+        # units required by the C++ layer below.  Also note that profile recentering for even-sized
+        # images (see the ._adjust_offset step below) leads to automatic reduction of stepK slightly
+        # below what is provided here, while maxK is preserved.
+        if calculate_stepk:
+            if calculate_stepk is True:
+                im = self.image
+            else:
+                # If not a bool, then value is max_stepk
+                R = (jnp.ceil(jnp.pi / calculate_stepk)).astype(int)
+                b = BoundsI(-R, R, -R, R)
+                b = self.image.bounds & b
+                im = self.image[b]
+            thresh = (1.0 - self.gsparams.folding_threshold) * self._image_flux
+            # this line appears buggy in galsim - I expect they meant to use im
+            R = _calculate_size_containing_flux(im, thresh)
+        else:
+            R = max(*self.image.array.shape) / 2.0 - 0.5
+        return self._getSimpleStepK(R)
+
+    def _getSimpleStepK(self, R):
+        # Add xInterp range in quadrature just like convolution:
+        R2 = self._x_interpolant.xrange
+        R = jnp.hypot(R, R2)
+        stepk = jnp.pi / R
+        return stepk
+
+    def _getMaxK(self, calculate_maxk):
+        if calculate_maxk:
+            _uscale = 1 / (2 * jnp.pi)
+            _maxk = self._x_interpolant.urange() / _uscale
+
+            if calculate_maxk is True:
+                maxk = _find_maxk(
+                    self._kim, _maxk, self._gsparams.maxk_threshold * self.flux
+                )
+            else:
+                maxk = _find_maxk(
+                    self._kim, calculate_maxk, self._gsparams.maxk_threshold * self.flux
+                )
```

Comment:
this blook always only uses the first `if` block?

Reply:
calculate_maxk can be a floating value or a bool in galsim.